### DISCUSSION TITLE
How to get response body using E2E with steps?

### DISCUSSION THREAD

**Discussion by odonnell-anthony** [pactumjs/pactum#297] [General]
Not using E2e I can get at the response body like so

```js
const response = await spec()
...
const responseBody = JSON.parse(response.text);
```
But using E2e and steps that does not work, it comes back as undefined

```js
describe('Do something', () => {

    let doer = e2e('Doing things');

    it('does stuff', async () => {
        const response = await doer.step('doing')
            .spec()
            ...
            const responseBody = JSON.parse(response.text)
```
How to get at the response body using E2E and steps?

**Comment by ASaiAnudeep** [comment]
Potential problem could be [this](https://github.com/pactumjs/pactum/blob/master/src/models/Step.js#L50). Instead of awaiting we might need to return it.
